compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/KotlinPoet.kt

```diff
@@ -173,15 +177,41 @@ private fun TypeName.wrapInLazy(): ParameterizedTypeName {
   return Lazy::class.asClassName().parameterizedBy(this)
 }
 
-internal fun TypeSpec.Builder.addAnvilAnnotation(): TypeSpec.Builder {
+private fun ModuleDescriptor.getGeneratedAnnotationClassName(): String? {
+  val targetVersion = (module.platform
+      ?.firstOrNull { it is JdkPlatform } as? JdkPlatform)
+      ?.targetVersion
+      ?: return null
+
+  return when {
+    targetVersion > JVM_1_8 -> "javax.annotation.processing.Generated"
+    targetVersion == JVM_1_8 -> "javax.annotation.Generated"
+    else -> null
```

Comment:
Will this return `null` for Android modules that target 1.8? On Android the annotation is never available. Dagger does not emit it in Android projects at all.

Reply:
I'd assume it wouldn't. Do you know how Dagger makes this distinction by any chance?

Reply:
It's checking the classpath https://github.com/google/auto/blob/master/common/src/main/java/com/google/auto/common/GeneratedAnnotations.java

Reply:
I definitely can reproduce what you said.

Reply:
Resolved.

Reply:
Great, thank you